### DISCUSSION TITLE
Ability to duplicate a collection or create collection templates

### DISCUSSION THREAD

**Discussion by iamkeir** [directus/directus#3414] [Feature Requests]
Posting this by request from [Discord message](https://discord.com/channels/725371605378924594/741330910594465823/788078102088646688).

I am working on a site with several pages that need identical (or almost identical) collections - creating each one individually is time-consuming and prone-to-error (it is easy to miss a setting that should match the other collections). Equally, if a change is required, it has to be manually updated for each collection.

Two possible ways to improve this:

1) Ability to create some kind of collection 'template' or preset, that can act as a starting point for a new collection. Even better if you could keep collections in-sync with the template so any changes made to template were mirrored in all collections using that template - although this could present challenges around when you want to duplicate and break-off into a bespoke collection, or when you want to keep-in sync.

2) Ability to duplicate an existing collection to use as a starting point for a new collection (much like the duplicate field action).

Appreciate this could be a quite a complex dance for relationships/fields/etc. but thought worth capturing none-the-less, and see if any discussion/ideas come from it.

P. S. Directus is RAD.

**Comment by benhaynes** [comment]
Yours is more comprehensive, but this other issue is related: https://github.com/directus/directus/discussions/2862

**Comment by vuchl** [comment]
I am voting for this too!

**Reply by rijkvanzanten** [reply]
@vuchl Do you vote for "duplicating a collection" or the more involved "template" idea? 🙂

**Reply by haniefss** [reply]
how about when duplicating there's option whether to duplicate the collection or to save as template? 🙂

**Reply by jamescammarano** [reply]
> how about when duplicating there's option whether to duplicate the collection or to save as template? 🙂


Sounds like scope creep ;)
I think templates would be coooool make me feel like I'm using Ableton but duplicating collections sounds like MVP.

**Reply by iamkeir** [reply]
@jaycammarano I agree, duplication seems like the simpler of the two.

**Reply by rijkvanzanten** [reply]
What would be the difference between a template, and a regular collection (that you won't use for anything, maybe hidden)? 🤔 

(besides the _could keep collections in-sync with the template so any changes made to template were mirrored in all collections using that template_)

**Reply by vuchl** [reply]
Hello folks!

Nice this gains some traction! I am voting for collection templates. Duplicating content is pretty fast with copy&paste or the API. But it would be an amazing feature too!

**Reply by chris-saunders-rocketlab** [reply]
I think templates would be far superior but my use case is different, I want to create not collection templates but collection item templates.

I'm yet to test this, but I think:

- if I created a collection with four 'template' items I would then make them non-editable (to non-admins) except for a field such as name (without editing in at least some way you do not get the option for duplication).
- I suspect the item duplicated via Save as Copy, would also be non-editable and the solution breaks down there. 
- Archiving does not look to be of any use here, archived items are also editable.

I'm thinking of doing something along the lines of:

- Add a status of Template.
- Templates can only be created by admins (only admins can change a collection item status to Template).
- Templates can only be saved as copies.

I don't think my purposes would require anything more, this is full templating for me.
It could be taken further, for example in terms of which roles can utilise the Template status but the above would be enough for a start.

**Reply by rasco** [reply]
Couldn't this whole thing be broken down into a few way simpler things?
When I think about it, what we want is really just:
- a **schema snapshot** (like the one that already exists), but only of **specific fields** 
- the ability to choose which fields to create the snapshot from
- the ability to apply that snapshot to one or multiple existing collections
- (or creating a new collection based on the schema snapshot)
- if needed, there could be a part in the app where the snapshots are saved as "templates", but a simple CLI tool would work just as well

**Reply by claytongulick** [reply]
@vuchl Can you describe in a little more detail what you mean by copy and paste for duplicating a collection?

We have the need to make a lot of duplicated (similar) collections to segregate client health care data. 

Collection templates would definitely solve the issue for us, but I don't see a manual way to duplicate a collection with all the settings etc... currently.

I'd be good with just a "import/export schema" that dumped/imported the json for a single collection for right now.

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by sandros94** [reply]
### Potential use case
I'm studying how to create a fullstack boilerplate to programmatically deploy a Nuxt+Directus instance, via docker-compose/helm, that would take advantage of the frontend components and the backend builder(m2a).

### My ideas so far
For normal deployments it would be enough to use `npx directus schema apply -yes schema.yml`.
But for smaller multi-site deployments, where spinning up a directus instance for each site isn't worth (eg: landing page+blog+ecommerce) it would be easier to handle somesort of collection templating and using a list of objects to define what template and collection prefix to use and how many times.
This would be easily manageable via docker env once the `schema-template.yml`(s) is(are) defined.

This is actually something I started discussing on discord.

P.S.: I have no idea how to manage migrations, I'm not yet at that point.

**Reply by sandros94** [reply]
Oh, just in case someone asks "why would you need to host multiple frontends that points to the same backends" I'll respond kindly:

"isn't this one of the points for **HEADLESS** CMSes?".

I can easily see how small startups would much benefit on such boilerplates to setup a single platform to handle both web content and internal ERP.

**Reply by yachal-upson** [reply]
Same questions on my mind. Sad not to see more movement on this thread. Did I miss a new feature?
